docs/explanations/design-old.rst

```diff
@@ -34,18 +34,19 @@ wheel (binary distribution) from Meson_ project.
 Source distribution (sdist)
 ---------------------------
 
-The source distribution is based on ``meson dist``, so make sure all your files
-are included there. In git projects, Meson_ will not include files that are not
-checked into git, keep that in mind when developing. By default, all files
-under version control will be included in the sdist. In order to exclude files,
-use ``export-ignore`` or ``export-subst`` attributes in ``.gitattributes`` (see
-the ``git-archive`` documentation for details; ``meson dist`` uses
-``git-archive`` under the hood).
-
-Local (uncommitted) changes to files that are under version control will be
-included. This is often needed when applying patches, e.g. for build issues
-found during packaging, to work around test failures, to amend the license for
-vendored components in wheels, etc.
+The source distribution is base on ``meson dist``: the archive is
+created adding the required metadata files to the archive obtained
+running ``meson dist --no-tests --allow-dirty``.  This archive
+contains the full contents of the latest commit in revision
+control.  All revision control metadata is removed.
+
+The ``meson dist`` uses the archival tool of the revision control
+system for creating the archive. Files can be excluded from the source
+distribution To exclude files from the source distribution, use
```

Comment:
A half-finished and duplicate sentence here. I'd remove "Files can ..." and keep "To exclude ..."